### ISSUE TITLE
[BUG] B2B: Blind transfer on multi-interface setup: buffer overrun, dropping

### ISSUE THREAD

**Issue by devoxy1** [OpenSIPS/opensips#3205]
Hello!

I have opensips server with 2 TCP interfaces and b2b module setup. I have side A calling to opensips, opensips creates new leg with new callid from second interface to B, then B sends REFER with `Refer-To:` to C. 

Here is part of my config file which processes REFER and creates new leg: 

```
if ($rm != "REFER") {
    # for requests other than REFER, no special actions needs to be done,
    # just pass the request to the peer
    xlog("L_INFO", "[$ci] [$rm] [B2B_LOGIC_REQUEST] Passing request - From-tag: $ft, To-tag: $tt");
    b2b_pass_request();
    exit;
}

b2b_send_reply(202, "Accepted");

$socket_out=$socket_in;
force_send_socket("$socket_out");

b2b_client_new("referee", $var(refer-to),,,, $avp(b2b_hdrs), $avp(b2b_hdr_bodies));
$rtp_relay = "replace-origin replace-session-connection";
$rtp_relay_peer = "replace-origin replace-session-connection";
#rtp_relay_engage("rtpengine");

# bridge the referrer's peer with the referee
b2b_bridge("peer", "referee",,"rollback-failed,notify,no-late-sdp");
```

**Opensips Version**

```
*version: opensips 3.4.1 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: c03bbc4f6
main.c compiled on 12:37:46 Sep 25 2023 with gcc 9
```

**Describe the bug**
The whole scenario works fine except for last BYE request, once C picks up the call opensips sends NOTIFY with final response to B and then sends BYE, in the mean time B upon receiving NOTIFY sends BYE as well. Once opensips receives BYE from B it respnses with 200 OK and then trying to send another BYE from second interface to first interfaces and these messages are looped between ints until buffer overruns and ends up with error:

`ERROR:core:tcp_read: buffer overrun, dropping`

Interesting observation is that if i remove `no-late-sdp` flag in b2b_bridge - the issue no longer appears and call ends up correctly.

**To Reproduce**
Example:
1. Call from A to opensips
2. Opensips creats new leg to B
3. B sends REFER with `Refet-to` to C
4. Opensips creates new leg to C
5. opensips send NOTIFY with call result and BYE to B
6. B sends BYE to opensips
7. Opensips cycles last BYE request and overrun TCP buffer

**Expected behavior**
Opensips send BYE and discard BYE(or sends 481) from B if dialog is already ended.

**Relevant System Logs**
`ERROR:core:tcp_read: buffer overrun, dropping
ERROR:core:tcp_read_req: failed to read `

**OS/environment information**
 - Operating System: Ubuntu 20.04.5 LTS
 - OpenSIPS installation: git
 - other relevant information:

**Additional context**
Here sngrep dialgram to reflect the problem
![image](https://github.com/OpenSIPS/opensips/assets/102153411/6ad3c4a7-2b83-44f0-859f-74c82bb31724)

**Comment by bogdan-iancu** [comment]
@devoxy1 , could you off-ticket share a pcap please ? If so, please label each IP involved in the trace.

**Comment by devoxy1** [comment]
Hello @bogdan-iancu,
How do i share? Due to BYE cycle the file is 56mb.

**Comment by bogdan-iancu** [comment]
Normally you could simply email me to bogdan at opensips dot org, but 56 M is too large indeed - can you crop out the crazy BYE-ing ?

**Comment by devoxy1** [comment]
@bogdan-iancu I just sent an email with details. Please let me know if you received and everything is ok.

**Comment by bogdan-iancu** [comment]
got it, let me check

**Comment by bogdan-iancu** [comment]
Thanks for the info @devoxy1 , I think I managed to understand the issue here - the received BYE is handled just after the B2B sent its BYE - so the received BYE finds the B2B dialog terminated, and it fails to match any ongoing dialog. Because of that, OpenSIPS is trying to the proxy like routing for the BYE, leading to that weird behavior . I will check to see what can be done here

**Comment by devoxy1** [comment]
@bogdan-iancu Thanks for looking into it. 
Some time ago I created feature request  which relates to this problem (opensips and remote side both send BYE after transfer is successful), so may be it can be considered as well - https://github.com/OpenSIPS/opensips/issues/3118

**Comment by bogdan-iancu** [comment]
@devoxy1 , please test the patch I just committed (see above) and let me know if it does the trick for you. IF everything ok, I will do the backport.

**Comment by devoxy1** [comment]
Hello @bogdan-iancu! Just made a test and the issue no longer appears. Thanks a lot!

**Comment by bogdan-iancu** [comment]
Do you see a 481 reply being sent out by b2b for the incoming BYE (the one that was looping) ?

**Comment by devoxy1** [comment]
No, both parties just send BYE and received reply 200OK. There is no 481 on last BYE

<img width="783" alt="Снимок экрана 2023-10-05 в 13 31 48" src="https://github.com/OpenSIPS/opensips/assets/102153411/e8efd8b4-266c-4e71-8735-7ca1eae467f8">
